src/analyzing/CFG_build.ml

```diff
@@ -240,41 +247,91 @@ let rec cfg_stmt : state -> F.nodei option -> stmt -> cfg_stmt_result =
   | Return (tok, e) ->
       let newi = state.g#add_node { F.n = F.NReturn (tok, e) } in
       state.g |> add_arc_from_opt (previ, newi);
-      state.g |> add_arc (newi, state.exiti);
+      state.g |> add_arc (newi, state.return_destination);
+      state.may_return := true;
       CfgFirstLast (newi, None)
-  | Try (try_st, catches, finally_st) ->
+  | Try (try_st, catches, else_st, finally_st) ->
```

Comment:
This change doesn't seem to be straightforward. It looks like we'll need thorough testing to ensure the new logic works correctly. Do we currently have a unit test for the CFG building code?

Reply:
We don't I think. The test plan is usually "run -cfg_il and looks like it's ok".
Maybe we should start some test infra there (in a separate PR).

Reply:
Currently, some of the tainting and constant propagation unit tests rely on the correctness of the CFG.

I'm currently writing some more unit tests that exercise these new arcs. Once I have those, I'll update this PR again, and I think it will be ready for review.

If a separate infra is useful, then that would be in a separate PR.

Thanks for all of your early feedback!